src/main/java/org/openstreetmap/atlas/geography/atlas/delta/AtlasDeltaGenerator.java

```diff
@@ -44,55 +70,134 @@ protected int onRun(final CommandMap command)
+                THREADS_SWITCH);
+    }
+
+    private void run(final Path before, final Path after, final Path outputDirectory)
+    {
+        final Time time = Time.now();
+
+        this.logger.info("Comparing {} and {}", before, after);
+
+        // If the after is a directory, we want to diff the individual shards in parallel.
+        if (Files.isDirectory(after))
+        {
+            // You need to have the before dir be a dir of shards too for this to work.
+            if (!Files.isDirectory(before))
+            {
+                logger.error("Your -before parameter must point to a directory of atlas shards if "
+                        + "you want to compare shard by shard with an -after directory also of shards!");
+                System.exit(COMMAND_LINE_USAGE_ERROR_EXIT);
+            }
+
+            // Execute in a pool of threads so we limit how many atlases get loaded in parallel.
+            final ForkJoinPool customThreadPool = new ForkJoinPool(threads);
+            try
+            {
+                customThreadPool
+                        .submit(() -> this.compareShardByShard(before, after, outputDirectory))
+                        .get();
+            }
+            catch (final InterruptedException interrupt)
+            {
+                logger.error("The shard diff workers were interrupted.", interrupt);
+            }
+            catch (final ExecutionException exec)
+            {
+                logger.error("There was an execution exception on the shard diff workers.", exec);
+            }
+        }
+        // Otherwise, we can do a normal compare where we look at 2 atlases or input shards with a
+        // single output.
+        else
+        {
+            final Atlas beforeAtlas = load(before);
+            final Atlas afterAtlas = load(after);
+            compare(beforeAtlas, afterAtlas, outputDirectory);
+        }
+
+        logger.info("AtlasDeltaGenerator complete. Total time: {}.", time.elapsedSince());
+    }
+
+    /**
+     * Load a multi atlas if directory, otherwise load single atlas.
+     *
+     * @param path
+     *            An atlas shard directory or a single atlas.
+     * @return An atlas object.
+     * @throws IOException
+     *             Exception if loading the atlas directory does not work.
+     */
+    private Atlas load(final Path path)
+    {
+        return Files.isDirectory(path) ? this.loadAtlasDirectory(path) : this.loadSingleAtlas(path);
     }
 
-    private void compare(final Atlas beforeAtlas, final Atlas afterAtlas, final Path outputDir)
+    private Atlas loadSingleAtlas(final Path path)
     {
-        final String name = beforeAtlas.getName().split(".atlas")[0];
+        return new AtlasResourceLoader().load(new File(path.toFile()));
+    }
+
+    private Atlas loadAtlasDirectory(final Path path)
+    {
+        return new AtlasResourceLoader().load(fetchAtlasFilesInDirectory(path));
```

Comment:
It almost feels like this functionality should be part of `AtlasResourceLoader` : given a path, load the file or files there. Seems very reusable and common logic. If others agree, it may be worthwhile to add it there.

Reply:
I am fairly certain `AtlasResourceLoader` already does this. Do:

```java
new AtlasResourceLoader().load(new File(path.toAbsolutePath().toString()))
```

Reply:
Yep, remember writing this haha! Looks like we can get rid of some duplicated code here.